src/praisonai/praisonai/auto.py

```
@@ -649,42 +649,20 @@ def __init__(self, topic="Movie Story writing about AI", agent_file="test.yaml",
-            else:
-                logger.warning(f"""
-Tools are not available for {framework}. To use tools, install:
-    pip install "praisonai[{framework}]"
-""")
+        # Validate framework availability using adapter registry
+        from .framework_adapters.registry import FrameworkAdapterRegistry
+        
+        try:
+            adapter = FrameworkAdapterRegistry.get_instance().create(framework)
+        except ValueError as e:
+            raise ImportError(f"Unknown framework '{framework}'. Available frameworks: {', '.join(FrameworkAdapterRegistry.get_instance().list_available())}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)